### PR TITLE
code refactoring | added prometheus metrics | activated deployment co…

### PR DESCRIPTION
**Pull request by ironoa**
…nfig for kusama and polkadot

### CODE REVIEW

File: helmfile.d/config/kusama/registrar-watcher-values.yaml.gotmpl [hunk 1]
```diff
@@ -2,7 +2,7 @@ config:
   node:
     endpoint: "wss://rpc-node-base-services-external-websockets/"
   registrar:
-    index: 0
+    index: 100
```
Comment (ironoa) [inline helmfile.d/config/kusama/registrar-watcher-values.yaml.gotmpl:5]: waiting for the real one

**Review by pampatzoglou — APPROVED** [review]
LGTM

**Review by fgimenez — APPROVED** [review]
Looks good, are you planning to add metrics later? So far you enable the metrics endpoint but there are no custom metrics there, prometheus is also not instructed to scrape it.

File: src/prometheus.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export class Prometheus {
+
+    constructor(private readonly logger: Logger) {}
+
+    startCollection(): void {
+        this.logger.info(
+            'Starting the collection of metrics, the metrics are available on /metrics'
+        );
+        promClient.collectDefaultMetrics();
```
Comment (ironoa) [inline src/prometheus.ts:14]: here we have a set of default ones

File: src/actions/start.ts [hunk 3]
```diff
@@ -1,49 +1,46 @@
...
+ 
+  const cfg = new Config<InputConfig>().parse(cmd.config);
+  
+  const logger = createLogger(cfg.logLevel);
+  
+  const server = _startHttpServer(cfg.port)
+
+  const promClient = new Prometheus(logger);
+  promClient.injectMetricsRoute(server)
+  promClient.startCollection()
```
Comment (ironoa) [inline src/actions/start.ts:38]: and here I'm collecting just these default metrics... it's also true that I've not set up any AlertRules so far

**Pull request merged**